platform/kobo/koreader.sh

```diff
@@ -218,28 +218,33 @@ while [ $RETURN_VALUE -ne 0 ]; do
         # Increment the crash counter
         CRASH_COUNT=$((CRASH_COUNT + 1))
         CRASH_TS=$(date +'%s')
-        echo "!!!!" >>crash.log 2>&1
-        echo "Hu oh, something went awry... (Crash n°${CRASH_COUNT}: $(date +'%x @ %X'))" >>crash.log 2>&1
-        if [ $CRASH_COUNT -le 5 ]; then
-            echo "Attempting to restart KOReader . . ." >>crash.log 2>&1
-            echo "!!!!" >>crash.log 2>&1
-        fi
 
         # Show a fancy bomb on screen
         viewWidth=600
-        eval "$(./fbink -e | tr ';' '\n' | grep -e viewWidth | tr '\n' ';')"
+        viewHeight=800
+        eval "$(./fbink -e | tr ';' '\n' | grep -e viewWidth -e viewHeight | tr '\n' ';')"
         # Given the (mostly) identical AR across all Kobos, a fraction of the *width* usually leaves us with something right above the center of the screen, so it doesn't clash with the boot progress bar ;)
         bombHeight=$((viewWidth/2 + viewWidth/4))
         bombMargin=$((viewWidth/30))
-        # With a little notice at the bottom of the screen (same line as KFMon), on a big gray screen of death ;).
-        ./fbink -q -b -c -B GRAY9 -m -y -5 "Don't Panic! (Crash n°${CRASH_COUNT} -> ${RETURN_VALUE})"
+        # With a little notice at the top of the screen, on a big gray screen of death ;).
+        ./fbink -q -b -c -B GRAY9 -m -y 1 "Don't Panic! (Crash n°${CRASH_COUNT} -> ${RETURN_VALUE})"
         # U+1F4A3, the hard way, because we can't use \u or \U escape sequences...
         # shellcheck disable=SC2039
         ./fbink -q -b -O -m -t regular=./fonts/freefont/FreeSerif.ttf,px=${bombHeight},top=${bombMargin} $'\xf0\x9f\x92\xa3'
+        # And then print the tail end of the log on the bottom of the screen...
+        crashLog="$(tail -n 25 crash.log)"
```

Comment:
Fancy, although if it's only for 2 seconds you might have to be a pretty fast reader. ;-)

Reply:
Oh, yeah, good point, delay should be higher now that it does this ;p.

Reply:
Yep :) Ideally, I'd like some button dialog on that, that waits for my choice: `Exit | Restart`, so we get the time to read it and decide. Can fbink do that ? :)

(After a crash, I usually read the crash.log with a KSM script in KSM own clumsy text viewer, or launch Koreader and view it in TextEditor - so that fbink display would avoid that need.)

Reply:
It's... tricky because of the various crappy input protocol handling on Kobos.

What might be doable is a *single* button, saying "Tap screen to <restart|exit KOReader>", the context 'i.e., restart or exit) being the position in the crash counter.
In which case we only need to listen for the first EV_KEY (or whatever) event that makes sense, and a tap anywhere would do the trick.

Reply:
If I'm able to just hack that counter to make my tap be always _too many already_, and make it always exit, I'm super fine with that :)

Reply:
I went with a "tap to continue" on the *first* crash in a series, which should workout well enough in practice ;).

Reply:
@poire-z: You raise an interesting conundrum: I *can* add an option in the Dev menu, but what it actually does is up for grabs:

I was originally thinking of simply reverting back to the previous behavior.
But I could also (and, in fact, it's probably going to be tiny bit easier) make it always print the log, wait for tap & *exit*.

Reply:
(essentially, check that dev setting, if it's on, always enforce the counter to 1, and at the end, just break).

Reply:
superfinewithme :)